src/core/dbus-execute.c

```diff
@@ -1573,7 +1573,12 @@ int bus_exec_context_set_transient_property(
                                 _cleanup_free_ char *str = NULL;
                                 size_t allocated = 0, len = 0, i, ncpus;
 
-                                ncpus = CPU_SIZE_TO_NUM(n);
+                                /* We assume that array received over D-Bus is cpu_set_t and hence its size must
+                                   be multiple of glibc's internal type. */
+                                if (n % sizeof(__cpu_mask) != 0)
```

Comment:
so this is problematic: we shouldn't leak glibc internal values over the bus. dbus clients might have different archs...

In fact, the whole cpuset code as it is is pretty borked, as it doesn't care about endianess issues, nor size of __cpu_mask...

This really needs to be cleaned up, so that we always generate the same payload here. Most likely we should reformat the array so that what we pass around here is always what glibc uses internally on x86-64/le, and then avoid any direct exposure of the types at all...